service/src/main/kotlin/fi/espoo/evaka/invoicing/service/NewDraftInvoiceGenerator.kt

```diff
@@ -366,45 +400,80 @@ class NewDraftInvoiceGenerator(private val productProvider: InvoiceProductProvid
+                InvoiceRow(
+                    id = InvoiceRowId(UUID.randomUUID()),
+                    periodStart = period.start,
+                    periodEnd = period.end!!,
+                    child = child,
+                    product = product,
+                    unitId = codes.unitId,
+                    amount = surplusAttendanceDays,
+                    unitPrice = calculateDailyPriceForInvoiceRow(price, dailyFeeDivisor),
+                )
+            )
+        } else listOf()
+    }
+
     private fun dailyAbsenceRefund(
         rows: List<InvoiceRow>,
-        period: DateRange,
         child: ChildWithDateOfBirth,
         absences: List<AbsenceStub>,
-        periodOperationalDates: List<LocalDate>,
-        attendanceDates: List<LocalDate>,
+        periodAttendanceDates: List<LocalDate>,
+        dailyFeeDivisor: Int,
         toInvoiceRow: (ProductKey, Int, Int) -> InvoiceRow
     ): List<InvoiceRow> {
         val total = invoiceRowTotal(rows)
         if (total == 0) return listOf()
 
-        val invoicePeriodTotalDateCount = attendanceDates.size
-
-        val refundedDayCount = getRefundedDays(period, child, absences, periodOperationalDates)
+        val invoicePeriodTotalDateCount = periodAttendanceDates.size
+        val refundedDayCount = getRefundedDays(child, absences, periodAttendanceDates)
         if (refundedDayCount == 0) return listOf()
 
         val (amount, unitPrice) =
             if (refundedDayCount >= invoicePeriodTotalDateCount) 1 to -total
-            else refundedDayCount to -calculateDailyPriceForInvoiceRow(total, periodOperationalDates.size)
+            else refundedDayCount to -calculateDailyPriceForInvoiceRow(
+                total,
+                minOf(periodAttendanceDates.size, dailyFeeDivisor)
+            )
 
         return listOf(toInvoiceRow(productProvider.dailyRefund, amount, unitPrice))
     }
 
     private fun getRefundedDays(
-        period: DateRange,
         child: ChildWithDateOfBirth,
         absences: List<AbsenceStub>,
-        periodOperationalDates: List<LocalDate>
+        periodAttendanceDates: List<LocalDate>
     ): Int {
         val forceMajeureAbsences = absences.filter { it.absenceType == AbsenceType.FORCE_MAJEURE }
-        val forceMajeureDays = forceMajeureAbsences.filter { period.includes(it.date) }
+        val forceMajeureDays = forceMajeureAbsences.filter { periodAttendanceDates.contains(it.date) }
 
         val parentLeaveAbsences = absences.filter { it.absenceType == AbsenceType.PARENTLEAVE }
         val parentLeaveDays = parentLeaveAbsences
             .filter { ChronoUnit.YEARS.between(child.dateOfBirth, it.date) < 2 }
-            .filter { period.includes(it.date) }
+            .filter { periodAttendanceDates.contains(it.date) }
 
-        return minOf(forceMajeureDays.size + parentLeaveDays.size, periodOperationalDates.size)
+        return minOf(forceMajeureDays.size + parentLeaveDays.size, periodAttendanceDates.size)
     }
```

Comment:
These changes currently break daily refunds for children in round the clock units. At least until planned absences are taken into account.

Reply:
Good catch, fixed